S3StorageWagon/src/main/java/com/gkatzioura/maven/cloud/s3/S3StorageRepository.java

```diff
@@ -150,25 +154,39 @@ public boolean newResourceAvailable(String resourceName,long timeStamp) throws R
     }
 
 
-    public List<String> list(String path) {
+    public List<String> list(String path) throws ResourceDoesNotExistException {
 
         String key = resolveKey(path);
 
         ObjectListing objectListing = amazonS3.listObjects(new ListObjectsRequest()
                     .withBucketName(bucket)
-                    .withPrefix(key));
+                    .withPrefix(key)
+                    .withDelimiter("/"));  // makes us traverse as if using directories
```

Comment:
The Wagon contract says the list operation should return the files and directories within the indicated directory, so I used the withDelimiter method, which causes S3 to behave like that.